### ISSUE TITLE
`isStandardSyntaxDeclaration` trusts the `extend` mark `postcss-less` reads out of any value holding the word

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#181]
Sibling of [#171](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/171), which drops the same kind of mark on the rule side. This one is about a declaration, and the mark reaches further: it is matched against the value of any declaration at all.

### What minimal example or steps are needed to reproduce the bug?

```less
a { b: "extend(x)" ,0; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-less",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/value-list-comma-space-before": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

`Unexpected whitespace before ","`, and `--fix` writing `a { b: "extend(x)",0; }` — what the same declaration gets in a `.css` file, since the text inside the quotes is a string and not a Less extend.

### What actually happened?

Nothing is reported and nothing is fixed. `postcss-less` marks the declaration:

```js
less.parse(`a { b: "extend(x)" ,0; }`).first.first.extend        // true
less.parse(`a { b: "myextend(x)" ,0; }`).first.first.extend      // true
less.parse(`a { content:"extend(x)"; }`).first.first.extend      // true
```

and `isStandardSyntaxDeclaration` reads that mark:

```js
// Less &:extend
if (`extend` in decl && decl.extend) return false
```

so every rule built on it passes the declaration over: `declaration-colon-newline-after`, `declaration-colon-space-before`, `declaration-colon-space-after`, `value-list-comma-newline-after`, `value-list-comma-newline-before`, `value-list-comma-space-after` and `value-list-comma-space-before`. Only a `.less` file is affected:

```
a { content:"extend(x)"; }   with declaration-colon-space-after: always
  .less   left as it stands
  .css    a { content: "extend(x)"; }
```

### Do you have a proposal to fix the bug?

The mark is put on by `LessParser`:

```js
// #123: add `extend` decorator to nodes
const extendPattern = /extend\(.+\)/i;

if (extendPattern.test(this.lastNode.value)) {
  this.lastNode.extend = true;
}
```

The pattern is anchored to nothing and reads the value as text, quotes and all, so it answers yes to a string holding the word, to a function whose name merely ends in it, and to any case at all — while Less reads `:extend` in lower case only, as the compiler shows by printing `.a:EXTEND(.b)` exactly as it stands.

What the mark is there for is the one shape `&:extend(.b);` — which `postcss-less` gives a declaration whose property is `&` and whose value opens with `:extend(`. Asking for that shape, rather than for the word anywhere in the text, is what the util needs; blanking the quoted runs before reading, as `isStandardSyntaxSelector` does since [#151](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/151), would do as well and would cost nothing else.

Worth reporting upstream too, as [#171](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/171) says of the rule-side mark: a `.extend` that answers "is this an extend" ought to answer it about the code rather than about the text.
